src/charm.py

```diff
@@ -179,6 +221,43 @@ def _on_provide_certificate_action(self, event: ActionEvent) -> None:
             return
         event.set_results({"result": "Certificates successfully provided."})
 
+    def _on_trust_certificate_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Provide trust certificate if configured to new requirer.
+
+        Args:
+            event: Juju event.
+
+        Returns:
+            None
+        """
+        try:
+            bundle = self._get_trusted_certificate_bundle()
+            self.certificate_transfer.add_certificates(
+                bundle,
+                relation_id=event.relation.id,
+            )
+        except ValueError as e:
+            logger.warning("Trust certificate relation cannot be fulfilled: %s", e)
+
+    def _get_trusted_certificate_bundle(self) -> Set[str]:
+        """Provide trust certificate if configured to new requirer.
+
+        Args:
+            event: Juju event.
+
+        Returns:
+            list[Certificate]: List of certificates in the bundle
+
+        Raises:
```

Comment:
Why to raise if the config is not provided? I understand when it is not valid.

Reply:
We can do a couple of things here:

1. Raise some kind of exception
2. Return an empty set
3. Return `None`

I think returning `None` is the worse option, as it makes the type checking more complex, and the code needs to validate that there is indeed an object there everywhere.

I tried returning an empty set, the control flow looked worse to me, as I had to handle exceptions, but also validate if the set was empty.

In the end I decided to let the lookup of the configuration raise `KeyError`. This makes the code even simpler as we never have to explicitly check the configuration, we do not pass `None` around, and we can still differentiate between the user not passing a value and the user passing a wrong value.